src/core/src/pass/constant_folding.cpp

```diff
@@ -96,12 +101,16 @@ bool ov::pass::ConstantFolding::run_on_model(const std::shared_ptr<ov::Model>& m
     return rewritten;
 }
 
-void ov::pass::ConstantFolding::copy_runtime_info_to_target_inputs(const std::shared_ptr<Node>& node,
-                                                                   const Output<Node>& replacement) {
-    for (auto& input : replacement.get_target_inputs()) {
-        auto consumer = input.get_node()->shared_from_this();
-        copy_runtime_info({node, consumer}, consumer);
+void ov::pass::ConstantFolding::copy_runtime_info_from_input_values(const std::shared_ptr<Node>& node) {
+    if (is_type<ov::opset1::ShapeOf>(node) || is_type<ov::opset3::ShapeOf>(node)) {
```

Comment:
Can we avoid this code for specific operations in common part?

Reply:
> Can we avoid this code for specific operations in common part?

There is no other way to determine it. Problem is in operation specification itself. `ShapeOf` is fused even their input is not Constant and this operation already has a special handling in `ConstantFolding`

Reply:
Moreover, it is code of transformation. I suppose we can check types of operations inside transformation)

Reply:
at least we need Base class for ShapeOf ops. 
Just to not update this line